pywt/_extensions/_pywt.pyx

```diff
@@ -794,28 +798,42 @@ cdef public class ContinuousWavelet [type ContinuousWaveletType, object Continuo
 
         """
         cdef pywt_index_t output_length "output_length"
-        cdef np.float64_t n
-        cdef psi_i, psi_r
-        cdef np.float64_t[::1] x, psi
+        cdef psi_i, psi_r, psi
+        cdef np.float64_t[::1] x64, psi64
+        cdef np.float32_t[::1] x32, psi32
 
-        n = pow(sqrt(2.), <double>level)
         p = (pow(2., <double>level))
         
         if self.w is not NULL:
             if length is None:
                 output_length = <pywt_index_t>p
             else:
                 output_length = <pywt_index_t>length
-            x = np.linspace(self.w.lower_bound, self.w.upper_bound, output_length)
-            # x = np.array(x, dtype=np.float64)
+            if (self.dt == np.float64):
+                x64 = np.linspace(self.w.lower_bound, self.w.upper_bound, output_length)
+            else:
+                x = np.linspace(self.w.lower_bound, self.w.upper_bound, output_length)
+                x32 = np.asarray(x, dtype=np.float32)
```

Comment:
I was going to suggest passing the dtype to `np.linspace`, but it looks like that was only added in `numpy` 1.9.  I have opened #189 to discuss potentially bumping up our minimum numpy version.